avocado/core/loader.py

```diff
@@ -618,18 +618,22 @@ def skip_non_test(exception):
                         tests.extend(self._make_tests(pth, which_tests))
         return tests
 
-    def _find_avocado_tests(self, path, class_name=None):
+    def _find_avocado_tests(self, path, class_name=None, instrumented=False):
```

Comment:
Yep, this sounds better 👍

Reply:
Unfortunately it's not enough. This is not per-discovery, but per-class variable. Anyway that didn't work before either, so it's not this patch fault, but still this solution changes the result and before it worked unexpectedly-correctly, now it behaves unexpectedly-incorrectly. The difference is when you have unittests and instrumented tests in a single file.

Let's use this example:

```python
$ cat avocado_and_unittest.py
import avocado
import unittest


class Avocado(avocado.Test):
    def test(self):
        pass


class Unit(unittest.TestCase):
    def test(self):
        pass
```

Original behavior was that if any instrumented test was present, only instrumented tests were reported, therefor only `Avocado` test was executed.

This patch discovers both, but claims they are both instrumented tests, therefor it fails with `Avocado crashed: TestError: Process died before it pushed early test_status.` trying to execute the `Unit` test.

The proper solution would be to report discovered unit-test-like tests (both, instrumented and unittest) and on success add both to the test-suite using the proper type.

Acceptable solution would be to keep the old behavior and first only discover instrumented tests and when none found discover unittests. But I think the amount of code to handle that would be similar (if not worse) compare the "proper" solution.